### PR TITLE
feat(type): optionally provide error type in response

### PR DESCRIPTION
**Pull request by tlhunter**
- adds two new configuration flags and a new dependency
  - https://github.com/sindresorhus/decamelize
- module remains backwards compatible
- since a new feature is being added, this is semver MINOR
- technically, since this no longer works with Node v8, it's semver MAJOR
  - the regex used by `decamelize` doesn't work with v8, but does as early as v10.0.0

### CODE REVIEW

**Comment by pdufour** [comment]
This is awesome! Great doc additions too. LGTM.

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 language: node_js
 sudo: false
 node_js:
-  - 8.11.3
+  - 10.0.0
```
Comment (mgartner) [inline .travis.yml]: Do they support 10.15.3 to match the version in `.node-version`?

File: lib/index.js [hunk 2]
```diff
@@ -103,6 +106,12 @@ exports.register = function (server, options, next) {
       }
     };
 
+    if (options.permeateErrorName) {
+      err.output.payload.error.type = options.decamelizeErrorName ?
+        Decamelize(err.name) : // MethodNotAllowed
+        err.name; // method_not_allowed
```
Comment (mgartner) [inline lib/index.js]: I think your comments are switched. `Decamelize(err.name)` should produce `method_not_allowed`, correct?

Comment (tlhunter) [reply lib/index.js]: You're right

File: .travis.yml (hunk 1 above)
Comment (tlhunter) [reply .travis.yml]: They should, I'll update

**Pull request merged**